lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,8 +451,20 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
 
     performTypoCorrection(DC, UDRE->getRefKind(), Type(), Name, Loc,
                           lookupOptions, Lookup);
+    
+    // Check if any lookups share a basename. If that's the case, we need
+    // to disambiguate the diagnostic by printing the UDRE's full name.
```

Comment:
In that case, this is probably okay for now, but please add a comment as to why it is necessary.